image/golem_blender_app/golem_blender_app/process_tools/__init__.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+import time
+
+from dataclasses import dataclass
+from golem_task_api.threading import Executor
+from psutil import Process
+
+
+@dataclass
+class Usage:
+    mem_peak: float = 0.  # bytes
+    cpu_time: float = 0.
+    real_time: float = 0.
+
+
+def _monitor_pid(pid: int, usage: Usage):
+    proc = Process(pid)
+
+    while proc.is_running() and not Executor.is_shutting_down():
+        usage.cpu_time = sum(proc.cpu_times())
+        usage.mem_peak = max(usage.mem_peak, proc.memory_info().vms)
+        time.sleep(0.5)
+
+
+async def exec_and_monitor_cmd(cmd):
+    usage = Usage()
+    time_started = time.time()
+
+    process = await asyncio.create_subprocess_exec(*cmd)
+    asyncio.ensure_future(Executor.run(_monitor_pid, process.pid, usage))
```

Comment:
Looks like there is a race condition here. Process is already running when the monitoring thread is spawned. However, I don't see how this could be improved. 😕

Reply:
It could be done in a more precise but much more complicated way. I believe these differences won't matter much in case of real world tasks